code/modules/mob/living/carbon/carbon.dm

```diff
@@ -1135,85 +1135,128 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
-	add_attack_logs(user, src, "Fed [toEat]. Reagents: [toEat.reagents.log_list(toEat)]", toEat.reagents.harmless_helper() ? ATKLOG_ALMOSTALL : null)
+	visible_message("<span class='warning'>[user] attempts to force [src] to swallow [to_eat].</span>")
+	if(!do_after(user, 3 SECONDS, TRUE, src))
+		return FALSE
+	forceFedAttackLog(to_eat, user)
+	visible_message("<span class='warning'>[user] forces [src] to swallow [to_eat].</span>")
+	return TRUE
 
+/mob/living/carbon/proc/forceFedAttackLog(obj/item/reagent_containers/to_eat, mob/user)
+	add_attack_logs(user, src, "Fed [to_eat]. Reagents: [to_eat.reagents.log_list(to_eat)]", to_eat.reagents.harmless_helper() ? ATKLOG_ALMOSTALL : null)
 
 /*TO DO - If/when stomach organs are introduced, override this at the human level sending the item to the stomach
 so that different stomachs can handle things in different ways VB*/
-/mob/living/carbon/proc/consume(obj/item/reagent_containers/food/toEat, bitesize_override, can_taste_container = TRUE)
-	var/this_bite = bitesize_override ? bitesize_override : toEat.bitesize
-	if(!toEat.reagents)
+/mob/living/carbon/proc/consume(obj/item/reagent_containers/food/to_eat, bitesize_override)
+	var/this_bite = bitesize_override ? bitesize_override : to_eat.bitesize
+	if(!to_eat.reagents)
 		return
 	if(satiety > -200)
-		satiety -= toEat.junkiness
-	if(toEat.consume_sound)
-		playsound(loc, toEat.consume_sound, rand(10,50), 1)
-	if(toEat.reagents.total_volume)
-		if(can_taste_container)
-			taste(toEat.reagents)
-		var/fraction = min(this_bite/toEat.reagents.total_volume, 1)
+		satiety -= to_eat.junkiness
+	if(to_eat.consume_sound)
+		playsound(loc, to_eat.consume_sound, rand(10, 50), TRUE)
+	if(to_eat.reagents.total_volume)
+		taste(to_eat.reagents)
+		var/fraction = min(this_bite / to_eat.reagents.total_volume, 1)
 		if(fraction)
-			toEat.reagents.reaction(src, toEat.apply_type, fraction)
-			toEat.reagents.trans_to(src, this_bite*toEat.transfer_efficiency)
+			to_eat.reagents.reaction(src, REAGENT_INGEST, fraction)
+			to_eat.reagents.trans_to(src, this_bite)
+
+/mob/living/carbon/proc/consume_patch_or_pill(obj/item/reagent_containers/p_or_p, user) // p_or_p = patch or pill
+	// The reason why this is bundled up is to avoid 2 procs that will be practically identical
+	if(!p_or_p.reagents.total_volume)
+		return TRUE // Doesn't have reagents, would be fine to use up
+
+	var/apply_method = "swallow"
+	var/reagent_application = REAGENT_INGEST
+	var/requires_mouth = TRUE
+	var/instant = FALSE
+	var/efficiency = 1
+
+	if(ispatch(p_or_p))
+		apply_method = "apply"
+		reagent_application = REAGENT_TOUCH
+		requires_mouth = FALSE
+		efficiency = 0.5 // Patches aren't that good at transporting reagents into the bloodstream
+		var/obj/item/reagent_containers/patch/patch = p_or_p
+		if(patch.instant_application)
+			instant = TRUE
+
+	if(user != src && !instant)
+		if(requires_mouth && (!head || !dna.species.dietflags)) // You will not feed the IPC
```

Comment:
don't IPCs have a port for drinking? feeding a pill through that makes more sense than a patch applying medicine through the metal chassis

Reply:
I went with the assumption of IPCs not being able to eat food. And patches are canonically piercing everything, including the heavy stuff like MODsuits and biohazard suits

Reply:
yeah but if the patch is piercing the chassis where does it go? IPCs dont have a bloodstream, meanwhile the connector port presumably goes to the posibrain or something